ingest-image/task.js

```diff
@@ -133,41 +124,53 @@ export default class Task {
     async save_image(md) {
         console.log(`Posting metadata to API: ${JSON.stringify(md)}`);
 
+        let image_id;
         try {
-            await fetcher(this.ANIML_API_URL, {
-                query: QUERY,
+            image_id = (await fetcher(this.ANIML_API_URL, {
+                query: `
+                    mutation CreateImageRecord($input: CreateImageInput!){
+                        createImage(input: $input) {
+                            image {
+                                _id
+                            }
+                        }
+                    }
+                `,
                 variables: {
                     input: {
                         md: md
                     }
                 }
-            });
+            })).data.createImage.image._id;
 
+            md._id = image_id;
             await this.copy_to_prod(md);
             await this.copy_to_archive(md);
         } catch (err) {
             if (err.message.includes('E11000')) err.message = 'DUPLICATE_IMAGE';
             console.log(`Error saving image: ${err}`);
 
-            await fetcher(this.ANIML_API_URL, {
-                query: `
-                    mutation CreateImageError($input: CreateBatchErrorInput!) {
-                        createImageError(input: $input) {
-                            _id
-                            batch
-                            error
-                            created
+            if (image_id) {
+                await fetcher(this.ANIML_API_URL, {
+                    query: `
+                        mutation CreateImageError($input: CreateBatchErrorInput!) {
+                            createImageError(input: $input) {
+                                _id
+                                batch
+                                error
+                                created
+                            }
+                        }
+                    `,
+                    variables: {
+                        input: {
+                            error: err.message,
+                            image: md.Hash,
```

Comment:
@ingalls - now that the image ID = the proejct ID + the image hash, shouldn't we be using `md._id` here?

Reply:
Fixed by using the returned Image ID from the API